Interaction States: NxCheckbox, NxRadio - RSC-1462 - RSC-1477

entroform opened the pull request:
https://issues.sonatype.org/browse/RSC-1462
https://issues.sonatype.org/browse/RSC-1477

avnvbhatta commented on gallery/visualtests/nxCheckbox.js:
```diff
@@ -37,8 +37,8 @@ describe('NxCheckbox', function() {
 
-    it(`has a blue background, white checkmark, a light blue outer border,
-      and glow when clicked and focused`, async function() {
+    it(`has a blue background, white checkmark, a blue outer border 
```
```suggestion
    it(`has a blue background, white checkmark, with a blue outer border 
```

avnvbhatta commented on gallery/visualtests/nxRadio.js:
```diff
@@ -40,8 +40,8 @@ describe('NxRadio', function() {
 
-    it(`has a thick blue border, white background, with a light blue outer border and glow
-      when clicked and focused`, async function() {
+    it(`has a thick blue border, white background, blue outer border '
```
```suggestion
    it(`has a thick blue border, white background, with a blue outer border '
```

avnvbhatta commented on lib/src/base-styles/_nx-radio-checkbox.scss:
```diff
@@ -39,8 +39,7 @@
       &::after {
-        border: 1px solid var(--nx-color-interactive-border-focus);
-        box-shadow: var(--nx-box-shadow-focus);
+        border: 2px solid var(--nx-color-interactive-border-focus);
```
I think we decided to use the `outline` property instead of a `border` for focus states.

EmilyBermudez88 replied:
> I think we decided to use the `outline` property instead of a `border` for focus states.

I personally think it doesn't hurt to leave it here, since it's a circle and would look a bit odd with a square focus outline for those (very few) stubborn Safari users. But I'm open to other opinions!

entroform replied:
> I think we decided to use the `outline` property instead of a `border` for focus states.

I also think it might be a good case to leave it as a border here for the same reason Emily mentioned, let me know your thoughts!

avnvbhatta commented:
I see that the older screenshots also had the focus states cut off. For the NxToggle ticket, I updated the visual test file so that the screenshot showed the entire focus region. I did something similar here: https://github.com/sonatype/sonatype-react-shared-components/blob/4f217bd9632ef0f9cf33f3bb7b5f598b2d371a8a/gallery/visualtests/nxToggle.js

EmilyBermudez88 commented:
![Screenshot 2023-02-28 at 9 16 14 AM](https://user-images.githubusercontent.com/89862727/221884036-2d4d31a9-a128-4e14-80de-b97b77a0ac68.png)
I seem to be seeing the radios & checkboxes with the focus border not centered? (Consistently off as I zoom in/out)

EmilyBermudez88 commented:
![Screenshot 2023-02-28 at 9 16 14 AM](https://user-images.githubusercontent.com/89862727/221885725-d42c4a4f-0864-4758-bb44-14c1ebb0c9a3.png)
It seems like the focus ring is off slightly on both the radio and checkbox as I zoom in/out (although seems to be ok in the snapshots, so maybe something I'm doing).  Does the positioning of the `&::after` need to be adjusted in `NxRadio.scss` & `NxCheckbox.scss` since the border has increased to 2px instead of 1px?

entroform commented:
> ![Screenshot 2023-02-28 at 9 16 14 AM](https://user-images.githubusercontent.com/89862727/221885725-d42c4a4f-0864-4758-bb44-14c1ebb0c9a3.png) It seems like the focus ring is off slightly on both the radio and checkbox as I zoom in/out (although seems to be ok in the snapshots, so maybe something I'm doing). Does the positioning of the `&::after` need to be adjusted in `NxRadio.scss` & `NxCheckbox.scss` since the border has increased to 2px instead of 1px?

Ooo this is weird, I can't seem to replicate this for some reason.
Which browser did you test this on?
I did find a bug where it's not showing at all on safari though, looking into that.

<img width="272" alt="Screenshot 2023-02-28 at 9 57 34 AM" src="https://user-images.githubusercontent.com/10160030/221891653-78ab6b0c-39b3-4304-830b-f625e29837ce.png">

entroform commented:
> ![Screenshot 2023-02-28 at 9 16 14 AM](https://user-images.githubusercontent.com/89862727/221885725-d42c4a4f-0864-4758-bb44-14c1ebb0c9a3.png) It seems like the focus ring is off slightly on both the radio and checkbox as I zoom in/out (although seems to be ok in the snapshots, so maybe something I'm doing). Does the positioning of the `&::after` need to be adjusted in `NxRadio.scss` & `NxCheckbox.scss` since the border has increased to 2px instead of 1px?

I was able to replicate this a little bit, It looks very subtle on my end, not like the screenshot you have (like one pixel off).
I think it wasn't showing on certain zoom increments.

It was also off (sometimes and annoyingly) on Safari, I'm not entirely sure how to fix it since:
the problem is with the browser interpolating the border sizing when you zoom in and out and rounding the rasterization.

This doesn't explain how it looks on your screenshot though, that looks completely off.
I haven't been able to replicate that yet, I'll play around a little bit more!

EmilyBermudez88 commented:
> I was able to replicate this a little bit, It looks very subtle on my end, not like the screenshot you have (like one pixel off). I think it wasn't showing on certain zoom increments.

That was my first thought as well, and tried in different zoom increments (and looked closely at the default 100%) and still seemed a bit off. I can touch base with you on this to see what's going on on my end

entroform commented:
> > I was able to replicate this a little bit, It looks very subtle on my end, not like the screenshot you have (like one pixel off). I think it wasn't showing on certain zoom increments.
> 
> That was my first thought as well, and tried in different zoom increments (and looked closely at the default 100%) and still seemed a bit off. I can touch base with you on this to see what's going on on my end

Oh! It was because I was editing the `scss` file inside the `/dist` folder haha
so it didn't get included in the commit.
That's why it looks okay on the snapshots.

Fixed here: 2ff66bdc44a5544cf2a8f9b558a957e2855ebdce

entroform commented:
> I see that the older screenshots also had the focus states cut off. For the `NxToggle` ticket, I updated the visual test file so that the screenshot showed the entire focus region. I did something similar [here](https://github.com/sonatype/sonatype-react-shared-components/blob/4f217bd9632ef0f9cf33f3bb7b5f598b2d371a8a/gallery/visualtests/nxToggle.js).

Thanks,
I updated the tests and snapshots here: 964c4d7242c3dbe85f5829705f6bdb98fd1d1aec

EmilyBermudez88 commented:
I noticed that some of the snapshots are of the entire example, and others are just of a single radio. What are your thoughts on having at least the disabled screenshots as just the radio you are trying to interact with? If the test is trying to hover or click the disabled radio, then it may be clearer which it's trying to interact with.

entroform commented:
> I noticed that some of the snapshots are of the entire example, and others are just of a single radio. What are your thoughts on having the disabled screenshots that are hovered or clicked as just the single radio ? If the test is trying to hover or click the disabled radio, then it may be clearer which it's trying to interact with.

I think that's a good idea, I'll wait until https://github.com/sonatype/sonatype-react-shared-components/pull/989 is merged so I can use the helper there.

JasonDSwearingen commented:
+1

entroform commented:
> I noticed that some of the snapshots are of the entire example, and others are just of a single radio. What are your thoughts on having the disabled screenshots that are hovered or clicked as just the single radio ? If the test is trying to hover or click the disabled radio, then it may be clearer which it's trying to interact with.

Updated it here: 249a0b5b54c63645134b7d4f00caefdce118d351

avnvbhatta commented on gallery/visualtests/nxRadio.js:
```diff
@@ -15,84 +17,94 @@ describe('NxRadio', function() {
   describe('Attribute-Disabled NxRadio', function() {
-    it('looks disabled by default', simpleTest(disabledSelector));
-    it('looks disabled when hovered', hoverTest(disabledSelector));
+    it(
+        'looks disabled when hovered',
+        hoverTest(disabledExampleLabelSelector, undefined, undefined, OUTSET)
+    );
```
The formatting is a bit off. I think it can fit in one line.

entroform replied:
> The formatting is a bit off. I think it can fit in one line.

Fixed here thanks!
: https://github.com/sonatype/sonatype-react-shared-components/pull/986/commits/df89b5ac89996ba8d249763625489ee14c687b49

avnvbhatta commented on gallery/visualtests/nxRadio.js:
```diff
@@ -15,84 +17,94 @@ describe('NxRadio', function() {
   describe('Attribute-Disabled-Checked NxRadio', function() {
-    it('looks disabled by default', simpleTest(disabledCheckedSelector));
-    it('looks disabled when hovered', hoverTest(disabledCheckedSelector));
+    it(
+        'looks disabled when hovered',
+        hoverTest(disabledExampleLabelCheckedSelector, undefined, undefined, OUTSET)
+    );
```
Same kind of formatting issue here.

entroform replied:
> Same kind of formatting issue here.

Fixed here thanks!: https://github.com/sonatype/sonatype-react-shared-components/pull/986/commits/df89b5ac89996ba8d249763625489ee14c687b49

avnvbhatta reviewed (approved):
🚀
